### PR TITLE
Feature/50 update frontend so that a displayname can be updated to an account

### PR DESCRIPTION
**Pull request by chenaicirt**
https://trello.com/c/PL6cXJMA/50-update-frontend-so-that-a-displayname-can-be-updated-to-an-account

### CODE REVIEW

File: src/app/_components/action-snack-bar-with-modal/snack-bar-component-example-snack.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<span>
+  Hello! Please set your display name. This is the name that will be shown when you post, comment, etc on the site, and will allow users to recognize who you are.
+</span>
+<button mat-raised-button
+  (click)="openDialog()"> Add Display Name </button>
+
+<!-- Copyright 2018 Google Inc. All Rights Reserved.
+    Use of this source code is governed by an MIT-style license that
+    can be found in the LICENSE file at http://angular.io/license -->
```
Comment (zaclittleberry) [inline src/app/_components/action-snack-bar-with-modal/snack-bar-component-example-snack.html]: @chenaicirt is there a reason there is license text in this file in particular?

Comment (chenaicirt) [reply src/app/_components/action-snack-bar-with-modal/snack-bar-component-example-snack.html]: Nope, this was just taken from the snackbar demo code that was on the material website

Comment (chenaicirt) [reply src/app/_components/action-snack-bar-with-modal/snack-bar-component-example-snack.html]: should I delete this?

File: src/app/_services/api.service.ts [hunk 2]
```diff
@@ -49,25 +49,24 @@ export class ApiService {
...
   }
 
   createAccount() {
     return this.httpClient.post<any>(`${this.apiUrl}/accounts`, '');
   }
 
-  getAccountLandlords() {
-    return this.httpClient.get<any>(`${this.apiUrl}/accounts/landlords`);
+  patchAccount(id, account: Account) {
+    return this.httpClient.patch<any>(`${this.apiUrl}/accounts/${id}`, account, {observe: 'response'});
```
Comment (zaclittleberry) [inline src/app/_services/api.service.ts]: the third param (object with observe response) is only needed if you need the full response (with headers) passed through. It doesn't look like you do, so this should be removed.

**Review by zaclittleberry — CHANGES_REQUESTED** [review]
There's a number of things in this pr to change. I'm not going to do each individually for this one, so apply the following where appropriate:
The longer bit of text I gave you was meant to go in the modal, not in the snackbar. I don't remember what the snackbar said before but I think that was fine as it was.
Please rename things (functions, classes, services, etc) semantically. so AccountDialog, AccountSnackBar, etc. I can tell you lifted code from examples, which is fine, but the code needs to be changed to be appropriate for this codebase. Once it's changed you can also remove the license at the bottom of that one file as it shouldn't be necessary in this case.
Please stick to style conventions and indent code properly, and multiple words in file names should be separated with dashes. Also services have a convention of ending in .service.ts, and their class name ending in Service. It is best to stick to any conventions that already exist in a codebase.
Also, I'm pretty sure you imported some classes into files they weren't used in, and I think may have imported a class into the same file twice. Please clean up any code that isn't actively in use; the only things committed should be what's necessary for the ticket at hand to work.

**Review by zaclittleberry — APPROVED** [review]

**Pull request merged**